lib/middleware/early_hints.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require_dependency "mobile_detection"
+
+module Middleware
+  class EarlyHints
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      @env = env
+      @request = ActionDispatch::Request.new(@env)
+      if hints_useful?
+        @request.send_early_hints("Link" => links)
+      end
+      @app.call(@env)
+    end
+
+    private
+
+    # Not compreensive but still useful
+    def hints_useful?
+      @request.get? &&
+      !@request.xhr? &&
+      !@request.path.ends_with?('robots.txt') &&
+      !@request.path.ends_with?('srv/status') &&
+      @request[Auth::DefaultCurrentUserProvider::API_KEY].nil?
+    end
+
+    def links
+      unless hints = Discourse.redis.get(hints_cache_key)
+        hints = hints_list
+        Discourse.redis.setex(hints_cache_key, hints_cache_duration, hints)
```

Comment:
Just curious, what is the rational for using a duration to break the cache instead of say something like `Discourse.git_version` since the asset manifests will only change on deploy?